pipelines/build/common/build_base_file.groovy

```diff
@@ -765,16 +765,33 @@ class Builder implements Serializable {
             tag = publishName
         }
 
+        if ((javaToBuild=="jdk21" || javaToBuild="jdk") && scmReference && !release) {
+            publishName = scmReference.replace('_adopt','')
+            def firstDot=publishName.indexOf('.')
+            def plusSign=publishName.indexOf('+')
+            def secondDot=publishName.indexOf('.', firstDot+1)
+            // Translate jdk-AA+BB to jdk-AA-0-BB
+            // Translate jdk-AA.B.C+DD to jdk-AA-C-DD-ea-beta
+            // Note that jdk-AA-B-C-D+EE will become jdk-AA-C-D-EE-ea-beta
+            if ( firstDot==-1 ) {
+                publishName=publishName.substring(4,plusSign)+'.0.'+publishName.substring(plusSign+1)
+            } else {
+                publishName=publishName.substring(4,firstDot)+publishName.substring(secondDot).replace("+","-")
+            }
+            publishName='ea_'+publishName.replaceAll("\\.","-")
+        }
+
         context.stage('publish') {
-            context.build job: 'build-scripts/release/refactor_openjdk_release_tool',
+        context.println "publishing with publishName: ${publishName}"
+        context.build job: 'build-scripts/release/refactor_openjdk_release_tool',
                     parameters: [
                         ['$class': 'BooleanParameterValue', name: 'RELEASE', value: release],
                         ['$class': 'BooleanParameterValue', name: 'DRY_RUN', value: ((releaseType=="Weekly" && javaVersion=="jdk21") ? true : false)],
```

Comment:
I guess it depends if we want to start regularly publishing weeklies for all releases.
Switch it to "no" except for JDK22 now since we'll have regular tags there? Also do you prefer a check on `javaVersion` or `javaToBuild`?